client/src/controllers/LoaderController.ts

```diff
@@ -0,0 +1,130 @@
+import { Application, Controller } from '@hotwired/stimulus';
+import { gettext } from '../utils/gettext';
+
+/**
+ * <button
+ *   type="submit"
+ *   class="button button-longrunning"
+ *   data-controller="w-loader"
+ *   data-w-loader-active-class="button-longrunning-active"
+ *   data-w-loader-active-value="{% trans 'Signing in…' %}"
+ *   data-w-loader-duration-seconds-value="40"
+ *   data-action="w-loader#activate"
+ * >
+ *  {% icon name="spinner" %}
+ *  <em data-w-loader-target="label">{% trans 'Sign in' %}</em>
+ * </button>
+ */
+export class LoaderController extends Controller {
+  static classes = ['active'];
+  static targets = ['label'];
+  static values = {
+    loading: { type: Boolean, default: false },
+    active: { type: String, default: gettext('Loading') },
+    durationSeconds: { type: Number, default: 30 },
+    label: { type: String, default: '' },
+  };
+
+  declare activeClass: string;
+  declare activeValue: string;
+  declare loadingValue: boolean;
+  declare durationSecondsValue: number;
+  declare labelTarget: HTMLElement;
+  declare hasActiveClass: boolean;
+  declare hasLabelTarget: boolean;
+  declare labelValue: string;
+  timer?: number;
+
+  static afterLoad(identifier: string, application: Application) {
+    const { controllerAttribute } = application.schema;
+    const { actionAttribute } = application.schema;
+
+    const updateLoader = () => {
+      const em = document.querySelector('em') as HTMLElement;
+      document
+        .querySelectorAll(
+          `.button-longrunning:not([data-controller~='${identifier}'])`,
+        )
+        .forEach((button) => {
+          button.setAttribute(controllerAttribute, identifier);
+          button.setAttribute(actionAttribute, 'w-loader#activate');
+
+          if (button.getAttribute('data-clicked-text')) {
+            button.setAttribute(
+              'data-w-loader-active-value',
+              `${button.getAttribute('data-clicked-text')}`,
+            );
+            button.removeAttribute('data-clicked-text');
+          }
+
+          if (em.matches('button em')) {
+            em.setAttribute('data-w-loader-target', 'label');
+          }
+
+          button.setAttribute('data-w-loader-duration-seconds-value', '10');
+        });
+    };
+
+    document.addEventListener('DOMContentLoaded', updateLoader);
+  }
+
+  activate() {
+    // If client-side validation is active on this form, and is going to block submission of the
+    // form, don't activate the spinner
+    const form = this.element.closest('form');
+
+    if (
+      form &&
+      form.checkValidity &&
+      !form.noValidate &&
+      !form.checkValidity()
+    ) {
+      return;
+    }
+
+    window.setTimeout(() => {
+      this.loadingValue = true;
+
+      const durationMs = this.durationSecondsValue * 1000;
+
+      // eslint-disable-next-line no-return-assign
+      this.timer = window.setTimeout(
+        () => (this.loadingValue = false),
+        durationMs,
+      );
```

Comment:
```suggestion
      this.timer = window.setTimeout(() => {
        this.loadingValue = false;
      }, durationMs);
```

If you put the update in a function `{}` body brackets  - you do not have to worry about the linting error.